src/Uno.UWP/Devices/Geolocation/Geolocator.Android.cs

```diff
@@ -11,113 +11,342 @@
+        }
+
+        public uint? DesiredAccuracyInMeters
+        {
+            get => _desiredAccuracyInMeters;
+            set
+            {
+                _desiredAccuracyInMeters = value;
+                RemoveUpdates();
+                RequestUpdates();
+            }
+        }
+
+        #region MyListener
+
+        public class MyLocListen : Java.Lang.Object, ILocationListener
+        {
 
-		public Task<Geoposition> GetGeopositionAsync(TimeSpan maximumAge, TimeSpan timeout)
-			=> GetGeopositionAsync();
+            public void OnLocationChanged(Location location)
+            {
+                DateTime date = new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc);
+                date = date.AddMilliseconds(location.Time);
+                if (date.AddMinutes(1) > DateTime.UtcNow)
+                { // only from last minute (we don't want to get some obsolete location)
+                    mbZmiany = true;
+                    moLocat = location;
+                }
+            }
 
-		public static async Task<GeolocationAccessStatus> RequestAccessAsync() => GeolocationAccessStatus.Allowed;
+            public void OnProviderDisabled(string provider)
+            {
+                throw new global::System.OperationCanceledException();
+            }
 
-		private LocationManager InitializeLocationProvider(double desiredAccuracy)
-		{
-			var locationManager = (LocationManager)global::Android.App.Application.Context.GetSystemService(Android.Content.Context.LocationService);
+            public void OnProviderEnabled(string provider)
+            {
+                
+            }
 
-			var criteriaForLocationService = new Criteria
-			{
-				Accuracy = Accuracy.Coarse
-			};
+            public void OnStatusChanged(string provider, Availability status, Bundle extras)
+            {
+                // This method was deprecated in API level 29 (Android 10). This callback will never be invoked.
+            }
+        }
 
-			var acceptableLocationProviders = locationManager.GetProviders(criteriaForLocationService, true);
 
-			if (acceptableLocationProviders.Any())
-			{
-				_locationProvider = acceptableLocationProviders.First();
-			}
-			else
-			{
-				_locationProvider = String.Empty;
-			}
+        MyLocListen mojSluchacz = new MyLocListen();
+        #endregion 
 
-			return locationManager;
-		}
+        public Geolocator()
+        {
+            _locationManager = (LocationManager)Android.App.Application.Context.GetSystemService(Android.Content.Context.LocationService);
 
-		partial void StartPositionChanged()
-		{
+            _reportInterval = 1000;
+            _movementThreshold = 0;
+
+            RequestUpdates();
+        }
+
+        ~Geolocator()
+        {
+            try
+            {
+                RemoveUpdates();
+            }
+            catch
+            { }
+
+            try
+            {
+                _locationManager.Dispose();
+            }
+            catch
+            { }
```

Comment:
What are the expected exceptions here? Why are they silenced?

Reply:
same as above (I'm not sure how works 'dispose' method and 'destructor' from same class).

Reply:
About dispose & finalize: https://www.c-sharpcorner.com/UploadFile/nityaprakash/back-to-basics-dispose-vs-finalize/

But the real question is: what is the lifetime of the `Geolocator` object? What happend if the application creates many instances of it? (because the constructor is `public`, they can).

For the `catch` here, once again, you need to assess the possible exceptions here.

Reply:
I didn't thought about "opening two Geolocator" case... So, maybe I should return to work from almost three months ago (and almost forgotten assumptions, test cases, etc.). But I think it would not be possible in this week.

Reply:
Ok. I think I recollect some memory from months ago...
LocationManager was static because it was used in RequestAccessAsync, one of version of this method. When it was only for GPS, and not for every location source.
But  bool _locChanged and Location _location have to be static, to be accessible from MyLocListen:OnLocationChanged.

I can un-static LocationManager, but, from the other hand, LocationManager is very power hungry on Android, and running two of them can drain battery very, very quickly.